src/amdgpu.cpp

```diff
@@ -149,17 +151,16 @@ void amdgpu_get_instant_metrics(struct amdgpu_common_metrics *metrics) {
 		int cpu_temp = 0;
 		if( IS_VALID_METRIC(amdgpu_metrics->temperature_core[0]) ) {
 			// prefered method
-			for (unsigned i = 0; i < cpuStats.GetCPUData().size() / 2; i++)
-				cpu_temp = MAX(cpu_temp, amdgpu_metrics->temperature_core[i]);
+			unsigned i = 0;
+			do cpu_temp = MAX(cpu_temp, amdgpu_metrics->temperature_core[i]);
+			while (i++ < ARRAY_SIZE(amdgpu_metrics->temperature_core) && IS_VALID_METRIC(amdgpu_metrics->temperature_core[i]));
 			metrics->apu_cpu_temp_c = cpu_temp / 100;
 		} else if( header->content_revision >= 3 && IS_VALID_METRIC(amdgpu_metrics->average_temperature_core[0]) ) {
 			// fallback 1
-			for (unsigned i = 0; i < cpuStats.GetCPUData().size() / 2; i++)
-				cpu_temp = MAX(cpu_temp, amdgpu_metrics->average_temperature_core[i]);
+			unsigned i = 0;
+			do cpu_temp = MAX(cpu_temp, amdgpu_metrics->average_temperature_core[i]);
+			while (i++ < ARRAY_SIZE(amdgpu_metrics->average_temperature_core) && IS_VALID_METRIC(amdgpu_metrics->average_temperature_core[i]));
 			metrics->apu_cpu_temp_c = cpu_temp / 100;
-		} else if( cpuStats.ReadcpuTempFile(cpu_temp) ) {
-			// fallback 2: Try temp from file 'm_cpuTempFile' of 'cpu.cpp'
-			metrics->apu_cpu_temp_c = cpu_temp;
```

Comment:
I removed this because it seemed like there were problems interpreting the `cpuStats` objects as the APU, but should this be kept in?

Reply:
The Ryzen 7000 iGPU requires this line of code to read and report the cpu temperature. This is because the `amdgpu_metric` file does not report any CPU temp for these CPUs, So I was forced to use the value reported by hwmon.

So I would like these line to remain if possible. What exactly is the problem you have with it? I don't really understand what you mean by:
> problems interpreting the `cpuStats` object as if it represented the APU

Maybe I can help looking for a solution?

Note that when using a dGPU you should never be able to reach this line of code, since there is an if statement that checks for `header->format_revision == 1`. See line 71 or 72 and 83 or 84.

Reply:
Oh ok! I'll add this back then.

Thanks! I don't really have that good of a context of how everything works in MangoHud 😅